horace_core/Tobyfit/detector_array.m

```diff
@@ -0,0 +1,78 @@
+function detectors = detector_array (wtmp, use_tubes)
+% Get detector information from a single sqw object.
+%
+%   >> detectors = detector_array (wtmp, use_tubes)
+%
+% This is a function for temporary use until the transition to exclusively
+% using IX_detector_array data in the experiment_info has been completed.
+%
+% Requires that each run in the sqw object has the same detpar structure or
+% IX_detector_array
+%
+% Input:
+% ------
+%   wtmp        sqw object
+%   use_tubes   Logical flag if detector information is not held in wtmp as an
+%               IX_detector_array object in wtmp.experiment_info:
+%               - true: use IX_det_He3tube with wall thickness 6.35e-4 m and 10
+%                 atms 3He partial pressures alongside detpar data on detector
+%                 sizes
+%               - false: use IX_det_TobyfitClassic as detector type alongside
+%                 detpar data data on detector sizes
+%
+% Output:
+% -------
+%   detectors   The single IX_detector_array object use for each of the runs
+
+
+% Because detpar only contains minimal information, either hardwire in
+% the detector type here or use the info now available in the detector
+% arrays
+detpar = wtmp.detpar();   % just get a pointer
+det = wtmp.experiment_info.detector_arrays;
+if isempty(det) || det.n_runs == 0
+    % no detector info was available when the sqw was populated, so
+    % continue with the old detector initialisation from detpar
+    if use_tubes
+        detectors = IX_detector_array (detpar.group, detpar.x2(:), ...
+            detpar.phi(:), detpar.azim(:),...
+            IX_det_He3tube (detpar.width, detpar.height, 6.35e-4, 10));   % 10atms, wall thickness=0.635mm
+    else
+        detectors = IX_detector_array (detpar.group, detpar.x2(:), detpar.phi(:), detpar.azim(:),...
+            IX_det_TobyfitClassic (detpar.width, detpar.height));
+    end
```

Comment:
I though we deprecate detpar and use `experiment_info.detector_arrays` everywhere. 

The code, presumably correctly, converts `detpar` to `det_array` but why is it necessary anywhere except the conversion of old file formats?

Reply:
@abuts Alex, you're correct that eliminating detpar is the ultimate goal, but it's still a work in progress. This PR gets us further towards it